### PR TITLE
CHE-768: add ability to delete multiple workspaces from workspace list

### PR DESCRIPTION
**Pull request by akurinnoy**
**_1 Upvote**_ Signed-off-by: Oleksii Kurinnyi okurinnyi@codenvy.com

### CODE REVIEW

**Comment by akurinnoy** [comment]
screenshot
@ashumilova @benoitf wdyt?

File: dashboard/src/app/workspaces/list-workspaces/list-workspaces.controller.js [hunk 1]
```diff
@@ -107,8 +123,104 @@ export class ListWorkspacesCtrl {
...
     }, (error) => {
-      console.log(error);
       this.isInfoLoading = false;
     });
   }
+
+  /**
+   * Delete all selected workspaces
+   */
+  deleteSelectedWorkspaces() {
```
Comment (benoitf) [inline dashboard/src/app/workspaces/list-workspaces/list-workspaces.controller.js:131]: is that code could be simplified by subcalls ? looks like we have a lot nested blocks

File: dashboard/src/components/api/che-workspace.factory.js [hunk 2]
```diff
@@ -317,10 +325,62 @@ export class CheWorkspace {
     }
 
     return 'ws://' + wsagentServerAddress + '/' + contextPath + '/ext/ws/' + workspaceId;
-
   }
 
   getIdeUrl(workspaceName) {
     return '/ide/' + workspaceName;
   }
+
```
Comment (benoitf) [inline dashboard/src/components/api/che-workspace.factory.js:306]: please add jsdoc for these methods of API : startUpdateWorkspaceStatus, addStatusListener and removeStatusListener

Also, the listener couldn't be setup automatically ?

Comment (benoitf) [inline dashboard/src/components/api/che-workspace.factory.js:306]: ok

File: dashboard/src/app/workspaces/list-workspaces/list-workspaces.controller.js (hunk 1 above)
Comment (benoitf) [inline dashboard/src/app/workspaces/list-workspaces/list-workspaces.controller.js:131]: ok

File: dashboard/src/app/workspaces/list-workspaces/list-workspaces.controller.js [hunk 3]
```diff
@@ -107,8 +121,97 @@ export class ListWorkspacesCtrl {
...
+        if (workspace.status === 'RUNNING') {
+          this.cheWorkspace.stopWorkspace(workspaceId);
+        }
+
+        // delete stopped workspace
+        stoppedStatusPromise
+          .then(() => {
+            return this.cheWorkspace.deleteWorkspaceConfig(workspaceId);
+          })
+          .then(() => {
```
Comment (ashumilova) [inline dashboard/src/app/workspaces/list-workspaces/list-workspaces.controller.js:171]: this `then` seems to belong to `deleteWorkspaceConfig`

**Comment by ashumilova** [comment]
+1

**Pull request merged**